### ISSUE TITLE
Accessibility: Clear button for product variations does not have an audible success message for screen reader users

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43611]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When using the clear button on the product single to reset selections on variable products, no success message is announced to screen reader users.

This video shows the clear button being triggered via keyboard and the variation fields clear but nothing is indicated to screen reader users that the button worked.
https://github.com/woocommerce/woocommerce/assets/13153456/e07c9d12-e343-4824-a16e-bfb0c8c5df88

### Expected behavior

There needs to be an auditory message for screen reader users so they know that the clear button works. 

Add a status message above or below the variation selection when options are cleared that says "Product selections cleared." Use [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) to ensure screen readers read the message when it appears. If you don't want the visual notice, you can hide it as screen reader only text.

### Actual behavior

See video shared above.

### Steps to reproduce

1. Turn on a screen reader such as VoiceOver or NVDA.
2. Go to a variable product and select options.
3. Use your mouse or keyboard to trigger the "clear" button.
4. Listen and hear that nothing is announced telling you that it was successful.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by dinhtungdu** [comment]
This can be resolved in one go with #43609. Thanks for the report!

**Comment by amesplant** [comment]
I will look into this issue this week.

**Comment by amesplant** [comment]
@dkotter this is ready for review ;)
